src/types.ts

```diff
@@ -1,12 +1,10 @@
-import BN from 'bn.js'
-
-export type Input = Buffer | string | number | bigint | Uint8Array | BN | List | null
+export type Input = string | number | bigint | Uint8Array | List | null
 
 // Use interface extension instead of type alias to
 // make circular declaration possible.
 export interface List extends Array<Input> {}
 
 export interface Decoded {
-  data: Buffer | Buffer[]
-  remainder: Buffer
+  data: Uint8Array | Uint8Array[]
```

Comment:
This is all looking awesome. Great updates.

if we want to fix #78 with this as well, should we define `data: Uint8Array | NestedUint8Array` with `export interface NestedUint8Array extends Array<Uint8Array | NestedUint8Array>`

Then decode can be simplified to:
`export function decode(input: Input, stream: boolean = false): Uint8Array | NestedUint8Array | Decoded`

if I'm understanding everything correctly

Reply:
Stream isn't implemented. What do you think if we merge the PR and then maybe work on others?

Reply:
ah ok, sounds good